app/scripts/modules/core/src/managed/ManagedResourceHistoryModal.tsx

```diff
@@ -99,6 +100,41 @@ const viewConfigurationByEventType = {
   },
 } as const;
 
+const mergeNewEvents = (next: IManagedResourceEvent[], previous: IManagedResourceEvent[]) => {
```

Comment:
This seems like an appropriate solve for this particular case because this is an event stream where the events are basically immutable.

We might need to start thinking about what we would do if that wasn't the case, i.e. a state of the world type of thing where the data we're looking at is more deeply nested and not immutable.

Reply:
yes strongly agree (as we've chatted about offline). I'd like to find a more appropriate and generalized solution than what we currently have in existing routes (e.g. executions) since those areas had angular compatibility as constraint and rely on all kinds of `.forceUpdate()` tomfoolery.